source/particles/data_out.cc

```diff
@@ -59,6 +111,140 @@ namespace Particles
     return dataset_names;
   }
 
+
+
+  template <int dim, int spacedim>
+  std::vector<
+    std::tuple<unsigned int,
+               unsigned int,
+               std::string,
+               DataComponentInterpretation::DataComponentInterpretation>>
+  DataOut<dim, spacedim>::get_nonscalar_data_ranges() const
```

Comment:
I hope that this is a 1:1 copy of the `DataOut_DoFData` function, right? Possibly shifting everything by one for the id.

Reply:
This is a 1:1 copy in which I removed all the code for complex output properties (because particles can not carry complex properties, unlike the DoF data).

Reply:
Ah, nice. Yes, that would have made things simpler...